src/interpreter/main.cpp

```diff
@@ -16,9 +16,18 @@
 #include "interpreter.hpp"
 #include "value.hpp"
 
-int main() {
+int main(int argc, char** argv) {
 
-	std::ifstream in_fs("source.jp");
+	if (argc < 2) {
+		std::cout << "Argument missing: source file" << std::endl;
+		return 0;
+	}
+
+	std::ifstream in_fs(argv[1]);
+	if (!in_fs.good()) {
```

Comment:
This will trigger if the file is empty, which you may or may not want to handle differently. (I'm ok with the way you did it, for the record)

The relevant documentation for the different failure modes of an `ifstream` can be seen here: https://en.cppreference.com/w/cpp/io/ios_base/iostate